src/Automation/SnapshotCommand.cs

```diff
@@ -35,28 +36,46 @@ public static ScanResults Execute(Config config, IScanTools scanTools)
             if (config.CustomUIAConfigPath != null)
                 scanTools.Actions.RegisterCustomUIAPropertiesFromConfig(config.CustomUIAConfigPath);
 
-            var rootElements = scanTools.TargetElementLocator.LocateRootElements(config.ProcessId);
+            List<ScanResults> resultList = new List<ScanResults>();
 
-            return scanTools.Actions.Scan(rootElements?.First(), (element, elementId) =>
+            var rootElements = scanTools.TargetElementLocator.LocateRootElements(config.ProcessId).ToList();
+
+            int targetIndex = 1;
+
+            foreach (var rootElement in rootElements)
             {
-                return ProcessResults(element, elementId, config, scanTools);
-            });
+                resultList.Add(scanTools.Actions.Scan(rootElement, (element, elementId) =>
+                {
+                    return ProcessResults(element, elementId, config, scanTools, targetIndex++, rootElements.Count);
+                }));
+            }
+
+            return resultList;
         }
 
-        private static ScanResults ProcessResults(A11yElement element, Guid elementId, Config config, IScanTools scanTools)
+        private static ScanResults ProcessResults(A11yElement element, Guid elementId, Config config, IScanTools scanTools, int targetIndex, int targetCount)
         {
             if (scanTools == null) throw new ArgumentNullException(nameof(scanTools));
             if (scanTools.ResultsAssembler == null) throw new ArgumentException(ErrorMessages.ScanToolsResultsAssemblerNull, nameof(scanTools));
 
             var results = scanTools.ResultsAssembler.AssembleScanResultsFromElement(element);
 
             if (results.ErrorCount > 0)
-                results.OutputFile = WriteOutputFiles(config.OutputFileFormat, scanTools, element, elementId);
+            {
+                if (targetCount > 0)
+                {
+                    results.OutputFile = WriteOutputFiles(config.OutputFileFormat, scanTools, element, elementId, (name) => $"{name}_{targetIndex}_of_{targetCount}");
+                }
+                else 
+                {
+                    results.OutputFile = WriteOutputFiles(config.OutputFileFormat, scanTools, element, elementId, null);
+                }
+            }
```

Comment:
@DaveTryon This is a modified version of your code. I am not sure if the target count query makes sense since this can never be smaller than 1. Should it be `targetCount > 1` to restore the existing behavior for single window scan file output?

Reply:
It should be `targetCount > 1`